### PR TITLE
#40790 Context switching, searching and bug fixes

### PR DESCRIPTION
**Pull request by manneohrstrom**
This is still going through design iterations but starting CR now.

Big pass on the panel to do polish, fix bugs and handle context switching. The highlights are:

- Improved search mode, including the hierarchy widget. Further work is needed here once we got the hierarchy search and entity resolve working
- Context buttons
- Retina support for top icons
- Added new official sg fields 
- formatting tweaks
- fixed bug in notes filters (!)
- added helper toolbar showing what you are looking at
- Included a my tasks on the project level, showing sq thumbs (default selected)

### CODE REVIEW

File: python/app/work_area_button.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+
+from sgtk.platform.qt import QtCore, QtGui
+import sgtk
+
+
+
+class WorkAreaButtonDetailsArea(QtGui.QToolButton):
```
Comment (josh-t) [inline python/app/work_area_button.py]: The class name feels odd. Should this be called something like `SetWorkAreaButton`?

File: python/app/work_area_button.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        if not self._is_current:
+            # collapse button after a delay
+            QtCore.QTimer.singleShot(300, self._init_default_state)
+
+        return super(WorkAreaButtonDetailsArea, self).leaveEvent(evt)
+
+
+
+
+class WorkAreaButtonListItem(WorkAreaButtonDetailsArea):
```
Comment (josh-t) [inline python/app/work_area_button.py]: Similar comment here about the class name.

File: python/app/work_area_button.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+    def enterEvent(self, evt):
+        """
+        QT Mouse enter event
+        """
+        if not self._is_current:
+            # not the current work area. so expand the button
+            self.setText(self._caption)
+            self.setToolButtonStyle(QtCore.Qt.ToolButtonTextBesideIcon)
+            self.setMinimumSize(QtCore.QSize(self._width, self.WIDGET_HEIGHT))
```
Comment (josh-t) [inline python/app/work_area_button.py]: Curious if you tried to use stylesheets to handle the hover look? I think I know the answer given our chat the other day about stylesheets. :)

Comment (manneohrstrom) [inline python/app/work_area_button.py]: hehe, yes, couldn't get it to work! qss seem to be a bit hit and miss, seems to work great with simpler "skinning" stuff but always with more complex stuff! i was trying to break out the "skinning" of the states by using custom properties, but the sizing i couldn't handle.

File: info.yml [hunk 4]
```diff
@@ -80,6 +86,6 @@ requires_engine_version:
 
 # the frameworks required to run this app
 frameworks:
-    - {"name": "tk-framework-shotgunutils", "version": "v4.x.x"}
+    - {"name": "tk-framework-shotgunutils", "version": "v5.x.x"}
```
Comment (manneohrstrom) [inline info.yml]: we already use v5 for the loader. do you see any increased risk in introducing in here? i guess we don't really have a choice if we want to use the hierarchy model...

File: python/app/work_area_button.py (hunk 3 above)
Comment (sgsteph) [inline python/app/work_area_button.py]: Just a comment about some _extreme_ qss styling examples: https://github.com/shotgunsoftware/tk-multi-importcut/blob/master/style.qss or https://github.com/shotgunsoftware/tk-multi-scriptedapp/blob/master/style.qss.

File: python/app/dialog.py [hunk 5]
```diff
@@ -163,10 +169,21 @@ def __init__(self, parent=None):
         self.ui.navigation_prev.clicked.connect(self._on_prev_clicked)
         
         # search
-        self.ui.search.clicked.connect(self._on_search_clicked)
-        self.ui.cancel_search.clicked.connect(self._cancel_search)
+        self.ui.browse.clicked.connect(self._enter_browse_mode)
+        self.ui.hierarchy_close.clicked.connect(self._exit_browse_mode)
```
Comment (manneohrstrom) [inline python/app/dialog.py]: fixed!

File: python/app/dialog.py [hunk 6]
```diff
@@ -818,26 +873,142 @@ def _on_prev_clicked(self):
...
-        self.ui.header_stack.setCurrentIndex(self.NAVIGATION_HEADER_IDX)
+        self.ui.main_stack.setCurrentIndex(self.NAVIGATION_MODE_IDX)
         self.ui.search_input.setText("")
-        
+
+    def _on_tree_doubleclicked(self, model_index):
+        """
+        Someone double clicked in the navigation tree
+        """
+        sg_item = shotgun_model.get_sg_data(model_index)
```
Comment (manneohrstrom) [inline python/app/dialog.py]: yes! how do i do the conversion? There didn't seem to be a simple way to do it?

File: python/app/model_hierarchy.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        """
+        # TODO - when tree system supports entity->path resolution, implement selection
+
+        if self._bundle.context.project:
+            # todo - refactor when we have the new entity resolution methods
+            path = "/Project/%s" % self._bundle.context.project["id"]
+        else:
+            path = "/"
+
+        if sgtk.util.get_published_file_entity_type(self._bundle.sgtk) == "PublishedFile":
```
Comment (manneohrstrom) [inline python/app/model_hierarchy.py]: much better! fixed

File: python/app/work_area_button.py (hunk 1 above)
Comment (manneohrstrom) [inline python/app/work_area_button.py]: YES! agree - sorry it's evolved through a series of refactors....

**Comment by manneohrstrom** [comment]
ready for another round @josh-t !

File: info.yml (hunk 4 above)
Comment (josh-t) [inline info.yml]: nope. i think just keep an eye out.

File: python/app/dialog.py (hunk 6 above)
Comment (josh-t) [inline python/app/dialog.py]: Here's how it's done in one of the demos: 

```
        # get the item from the source model
        selected_item = self._hierarchy_model.itemFromIndex(
            # get the source hierarchy model index
            self._hierarchy_proxy.mapToSource(model_index)
        )
```

**Pull request merged**